src/blockchain/mod.rs

```diff
@@ -110,6 +110,13 @@ pub trait GetTx {
     fn get_tx(&self, txid: &Txid) -> Result<Option<Transaction>, Error>;
 }
 
+#[maybe_async]
+/// Trait for getting block hash by block height
+pub trait GetBlockHash {
+    /// fetch block hash given its height
+    fn get_block_hash(&self, height: u64) -> Result<Option<BlockHash>, Error>;
```

Comment:
Here it seems like this method will return `Ok(None)` if the height is non-existent, `Ok(Some)` is everything went good, an Error in all the other cases. But then, when you implement it, you always either return an Error (using ?) or `Ok(Some(...))` - you never return `Ok(None)`.

Do you think it makes sense to change this method to:
```
fn get_block_hash(&self, height: u64) -> Result<BlockHash, Error> {
```

?

Reply:
Good point! Yeah I think it will be best to change the signature of the method as you suggested given that I never return `Ok(None)`